src/decoder/cuda-decoder-utils.h

```diff
@@ -0,0 +1,161 @@
+// decoder/cuda-decoder-utils.h
+
+// Copyright      2018  Zhehuai Chen
+
+// See ../../COPYING for clarification regarding multiple authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+// THIS CODE IS PROVIDED *AS IS* BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, EITHER EXPRESS OR IMPLIED, INCLUDING WITHOUT LIMITATION ANY IMPLIED
+// WARRANTIES OR CONDITIONS OF TITLE, FITNESS FOR A PARTICULAR PURPOSE,
+// MERCHANTABLITY OR NON-INFRINGEMENT.
+// See the Apache 2 License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef KALDI_CUDA_DECODER_UTILS_H_
+#define KALDI_CUDA_DECODER_UTILS_H_
+
+
+#include <cuda_runtime_api.h>
+#include "cuda_runtime.h"
+#include "util/stl-utils.h"
+#include "omp.h"
+#include <algorithm>
+#include <float.h>
+#include <math.h>
+#include <cooperative_groups.h>
+#include "math_constants.h"
+#include "omp.h"
```

Comment:
OpenMP is a sizable dependency. @dpovey Any thoughts on this?

Reply:
RE OpenMP: yes, I wasn't so happy about that.  There are still a lot of issues here that I don't fully understand; this code came originally from Justin Luitjens at NVidia.  When he has time I want to talk to him about this.